support/widgetConfigUtils/basemapToggle.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * This module contains a method to instantiate the 4.x API BasemapToggle Widget
+ * using configuration variable which come from the Config Panel.
+ */
+
+import { _findNode, esriWidgetProps } from "./widgetConfigUtils";
+import BasemapToggle from "esri/widgets/BasemapToggle";
+import Basemap from "esri/Basemap";
+
+//////////////////////////////
+// Public Module Functions
+//////////////////////////////
+
+/** The original basemap for the Webmap */
+let _originalBasemap: __esri.Basemap;
+/** The alternate basemap in the BasemapToggle */
+let _nextBasemap: __esri.Basemap;
+
+/**
+ * Adds BasemapToggle to Application, including logic to make 
+ * integrations with the Config Panel function properly
+ * @param props 
+ */
+export async function addBasemapToggle(props: esriWidgetProps): Promise<__esri.BasemapToggle> {
+
+    // Variables
+    const { view, config, propertyName, portal } = props;
+    const { basemapToggle, basemapTogglePosition, basemapSelector, nextBasemap } = config;
+    /** Depending on which is defined, use "basemapSelector"
+     * (which is an id, ex: db6a43e20e09486b9444edcd77e0f906) or "nextBasemap" 
+     * (which is a well-known-basemap-id: https://developers.arcgis.com/javascript/latest/api-reference/esri-Map.html#basemap).
+     * The "nextBasemap" config variable is kept for backward compatibilty with the Config Panel,
+     * whereas "basemapSelector" is the new config variable */
+    const alternateBasemapId: string = basemapSelector || nextBasemap;
+    let basemapToggleInstance: __esri.BasemapToggle;
+    
+    const node = _findNode("esri-basemap-toggle");
```

Comment:
I don't think we want to handle creating the Basemap Toggle in app base or adding it to the view. I think app base should. just be responsible for building the input props needed for the app to create the Basemap  Toggle.  So here we'll just have logic to get the proper values for the basemap. I think the example in config app examples should handle the rest of the logic of creating the widget and adding it to the view.

Reply:
Yeah I don't think we want to handle any widget initializations in the utility. It should just work with configuring the basemaptoggle widget. It should handle setup of any configuration only.

Reply:
I've refactored the implementation here. You'll see that the new `getBasemaps` function is the important function now, and it uses internal state in the module to keep track of the proper Basemaps to be used in the BasemapToggle. The rest of the logic is now in the configuration-ts example code